### PR TITLE
use lodash-fp rather than lodash

### PR DESCRIPTION
**Pull request by svozza**
As discussed in #17, I've refactored nearly all the functions of arity 1 to be points free using `lodash/fp`.

### CODE REVIEW

**Comment by nackjicholson** [comment]
Lookin real good! I'll have to pull it down and really dig into it. Thanks a lot for doing this, I like the approach

File: unmarshalItem.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 'use strict';
 
+var _ = require('lodash/fp');
```
Comment (nackjicholson) [inline unmarshalItem.js]: used?

File: marshalItem.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
 'use strict';
 
+var _ = require('lodash/fp');
```
Comment (nackjicholson) [inline marshalItem.js]: same, not used here either.

Comment (svozza) [inline marshalItem.js]: Yeah, I had been using it but I felt that it actually made the code less clear so I left it as it was.

**Comment by svozza** [comment]
I was going to add Haskell-like type sigs as comments at the top the functions but I wasn't sure how familiar you were with them. I'm more than happy to do so as they're quite helpful with point free code.

**Comment by nackjicholson** [comment]
Hindley-Milners??? try it dawg. Might not make any sense because we don't have type classes. You could kind of fake it I guess right?

```js
marshalNumber :: (Number a, NS b) => (a -> Boolean) -> (a -> b) -> b
```

Kind of strange since `NS` isn't really a thing, If it were I guess it would be a data constructor for a type called DynamoDbAttributeItem or something...

Play with it, but I think if your goal is to make it more clear, those signatures won't really accomplish that.

**Comment by notxt** [comment]
🐶

**Comment by bencassidy** [comment]
:dog:

**Comment by nackjicholson** [comment]
ICWUDT

**Comment by svozza** [comment]
Well I wasn't thinking of anything fancy, more just a way for people know what arguments to provide the function without having to check the `transform` function.

**Comment by notxt** [comment]
🆗

**Comment by notxt** [comment]
:shipit:

**Comment by nackjicholson** [comment]
Note, don't publish if merged. One of us needs to run a browser build and such.

**Pull request merged**